### ISSUE TITLE
ProteinPrepare re-prepare

### ISSUE THREAD

**Issue by GerardBCN** [Acellera/htmd#200]
As it is an feature still under development, I expect to find more than one bug until we get to the final result. I propose using this thread to report problems related to this function.

The first error I found is when I used the usual proteinPrepare on "protein and chain A" of PDBID 4UAI. It used to work before.

```
In [4]: [x, y] = proteinPrepare(prot)
2016-12-07 10:57:56,936 - propka - INFO - No pdbfile provided
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-4-4b9e10758c16> in <module>()
----> 1 [x, y] = proteinPrepare(prot)

/shared/gerard/HTMD/htmd/htmd/builder/preparation.py in proteinPrepare(mol_in, pH, verbose, returnDetails, hydrophobicThickness, holdSelection)
    349         logger.warning("The following residue has not been optimized: " + missedligand)
    350 
--> 351     mol_out, resData = _buildResAndMol(pdb2pqr_protein)
    352 
    353     mol_out.box = mol_in.box

/shared/gerard/HTMD/htmd/htmd/builder/preparation.py in _buildResAndMol(pdb2pqr_protein)
    127                             occupancy, beta, charge, record)
    128 
--> 129     resData._importPKAs(pdb2pqr_protein.pka_protein)
    130 
    131     return mol_out, resData

/shared/gerard/HTMD/htmd/htmd/builder/preparationdata.py in _importPKAs(self, pkaCont)
    158             chain = grp.atom.chainID
    159             icode = grp.atom.icode
--> 160             pos = self._findRes(resname, resid, icode, chain, forceAppend)
    161             self.data.set_value(pos, 'pKa', grp.pka_value)
    162             self.data.set_value(pos, 'buried', grp.buried * 100.0)

/shared/gerard/HTMD/htmd/htmd/builder/preparationdata.py in _findRes(self, a_resname, a_resid, a_icode, a_chain, forceAppend)
    120             pos = int(pos)
    121         else:
--> 122             assert False, "More than one resid matched (internal error, please report)"
    123         return pos
    124 

AssertionError: More than one resid matched (internal error, please report)
```

**Comment by j3mdamas** [comment]
You're using the HEAD of master, correct? I see @tonigi making some modifications to the code, so maybe he can comment before I look into it?

**Comment by GerardBCN** [comment]
Yes, @tonigi  should be assigned to this task.

**Comment by GerardBCN** [comment]
The tutorial is not functional on latest conda version (1.5.4):

```
In [16]: tryp = Molecule('3PTB')
2016-12-07 12:04:38,651 - htmd.molecule.molecule - INFO - Attempting PDB query for 3PTB

In [17]: tryp_op, prepData = proteinPrepare(tryp, returnDetails=True)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-17-21589925a583> in <module>()
----> 1 tryp_op, prepData = proteinPrepare(tryp, returnDetails=True)

/shared/gerard/anaconda3/lib/python3.5/site-packages/htmd/builder/preparation.py in proteinPrepare(mol_in, pH, verbose, returnDetails, hydrophobicThickness, holdSelection)
    266                                                              ph_calc_method="propka31",
    267                                                              ph_calc_options=propka_opts,
--> 268                                                              holdList=hlist)
    269     tmpin.close()
    270 

ValueError: too many values to unpack (expected 4)
```

**Comment by tonigi** [comment]
That's a separate problem. The current master requires htmd-pdb2pqr >= 2.1.2a9 , the previous ones require an older version. Maybe you are using  packaged htmd (=old) instead of master.

**Comment by GerardBCN** [comment]
True. Latest version of htmd-pdb2pqr is incompatible with latest conda htmd... I'll wait for you to fix the first problem then.

**Comment by tonigi** [comment]
duplicate assertion possibly fixed in abbdb70b45924f7f114e9cb36f483f9ecd0cb14e
